packages/components/src/components/as-category-widget/as-category-widget.spec.ts

```diff
@@ -0,0 +1,202 @@
+import { TestWindow } from '@stencil/core/dist/testing';
+import { Category, CategoryWidget } from './as-category-widget';
+
+describe('as-category-widget', () => {
+  let categoryWidget;
+
+  beforeEach(() => {
+    categoryWidget = new CategoryWidget();
+  });
+
+  it('should build', () => {
+    expect(new CategoryWidget()).toBeTruthy();
+  });
+
+  describe('Rendering', () => {
+    let element: HTMLAsCategoryWidgetElement;
+    let testWindow: TestWindow;
+
+    beforeEach(async () => {
+      testWindow = new TestWindow();
+      element = await testWindow.load({
+        components: [CategoryWidget],
+        html: '<as-category-widget></as-category-widget>'
+      });
+    });
+
+    it('should render properly', async () => {
+      element.heading = 'Category Widget Example';
+      element.description = 'Description for Category Widget';
+      element.categories = exampleCategories;
+      element.showClearButton = true;
+      await testWindow.flush();
+
+      expect(element).toMatchSnapshot();
```

Comment:
What is the snapshot that is compared to? Where?

Reply:
https://jestjs.io/docs/en/snapshot-testing

Reply:
That's basically it. It creates the references in a folder, which is commited to the repo, to be able to compare them in every test run. If there's something different, it fails.